worldedit-core/src/main/java/com/sk89q/worldedit/util/imagebrush/ImageBrushManager.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * WorldEdit, a Minecraft world manipulation toolkit
+ * Copyright (C) sk89q <http://www.sk89q.com>
+ * Copyright (C) WorldEdit team and contributors
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.sk89q.worldedit.util.imagebrush;
+
+import com.sk89q.worldedit.WorldEdit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Loads and caches image files from WorldEdit's brushes directory.
+ */
+public class ImageBrushManager {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private final Map<String, BufferedImage> images = new HashMap<>();
+    private final WorldEdit worldEdit;
+
+    public ImageBrushManager(WorldEdit worldEdit) {
+        this.worldEdit = worldEdit;
+    }
+
+    /**
+     * Loads all files from the brushes directory into the images cache.
+     */
+    public void load() {
+        images.clear();
+
+        File brushDir = new File(worldEdit.getPlatformManager().getConfiguration().getWorkingDirectory(), "brushes");
```

Comment:
I don't feel brushes is the correct directory for this. What if we wanted a model brush or a script brush in the future?

I'd personally prefer for this to be more generic